### PR TITLE
handle IM7 constant change

### PR DESCRIPTION
**Pull request by valentinxxx**
Fix for issue: #607 
IMGTYPE_TRUECOLORMATTE -> IMGTYPE_TRUECOLOR

### CODE REVIEW

File: src/Intervention/Image/Imagick/Decoder.php [hunk 1]
```diff
@@ -20,7 +20,7 @@ public function initFromPath($path)
 
             $core->setBackgroundColor(new \ImagickPixel('transparent'));
             $core->readImage($path);
-            $core->setImageType(\Imagick::IMGTYPE_TRUECOLORMATTE);
+            $core->setImageType(defined('\Imagick::IMGTYPE_TRUECOLORMATTE') ? \Imagick::IMGTYPE_TRUECOLORMATTE : \Imagick::IMGTYPE_TRUECOLOR);
```
Comment (samuel-cloete) [inline src/Intervention/Image/Imagick/Decoder.php]: Shouldn't it be `IMGTYPE_TRUECOLORALPHA`...? [https://github.com/mkoppanen/imagick/blob/master/imagick_helpers.c#L1184](https://github.com/mkoppanen/imagick/blob/master/imagick_helpers.c#L1184)

Comment (valentinxxx) [inline src/Intervention/Image/Imagick/Decoder.php]: @samuel-cloete you're absolutely right, thank you ;)

**Comment by olivervogel** [comment]
Thanks.

**Comment by Preau** [comment]
On ImageMagick 7.0.4-6 and imagick 3.4.2_7 I now get the following error:

Undefined class constant 'IMGTYPE_TRUECOLORALPHA'

**Comment by valentinxxx** [comment]
@Preau it seems it only works in imagick >= 3.4.3
https://github.com/mkoppanen/imagick/blob/3.4.3/ChangeLog#L33

**Comment by Preau** [comment]
@valentinxxx Thanks, I see a new formula for php70-imagick was released on Homebrew 2 days ago, upgrading solved my issue.

**Comment by Preau** [comment]
No it is broken < 3.4.3 actually, upgrading solved the issue.

**Comment by valentinxxx** [comment]
@olivervogel I think it's broken if the imagick version is < 3.4.3 and ImageMagick >= 7
https://github.com/mkoppanen/imagick/blob/3.4.2/imagick_helpers.c#L1153-L1155

**Comment by olivervogel** [comment]
Unfortunately I can't test it right now, but maybe this code will handle this issue better?

https://github.com/Intervention/image/pull/684/files

**Pull request merged**